rust/frontend/src/frontend.rs

```diff
@@ -313,16 +314,72 @@ impl Frontend {
         )
         .map_err(|err| match err {
             ToRecordsError::InconsistentLength => {
-                chroma_types::UpsertCollectionError::InconsistentLength
+                chroma_types::UpsertCollectionRecordsError::InconsistentLength
             }
         })?;
 
         self.log_client
-            .push_logs(collection_id, records)
+            .push_logs(request.collection_id, records)
+            .await
+            .map_err(|err| Box::new(err) as Box<dyn ChromaError>)?;
+
+        Ok(chroma_types::UpsertCollectionRecordsResponse {})
+    }
+
+    pub async fn delete(
+        &mut self,
+        request: chroma_types::DeleteCollectionRecordsRequest,
+    ) -> Result<
+        chroma_types::DeleteCollectionRecordsResponse,
+        chroma_types::DeleteCollectionRecordsError,
+    > {
+        let scan = self
+            .fetch_collection_snapshot(request.collection_id)
+            .await?;
+        let get_result = self
+            .executor
+            .get(Get {
+                scan,
+                filter: Filter {
+                    query_ids: request.ids,
+                    where_clause: request.r#where,
+                },
+                limit: Limit {
+                    skip: 0,
+                    fetch: None,
```

Comment:
is there something special we should do here? seems like the Python client ignores the potential issue here

Reply:
Is the issue that you can aggressively delete, leading to a buffer overrun from grpc?

Reply:
yes, or if there's a default limit we may not delete all the records the user expects